### PR TITLE
soc: intel_adsp/ace: wait for lpsram power up

### PR DESCRIPTION
**Pull request by abonislawski**
Wait for lpsram power up before bbzero

Fixes commit 195db14

### CODE REVIEW

**Review by mwasko — DISMISSED** [review]
@nashif please prioritize this fix - it is required to unblock MTL support

**Review by marcinszkudlinski — DISMISSED** [review]

File: soc/xtensa/intel_adsp/ace/sram.c [hunk 1]
```diff
@@ -40,10 +40,17 @@ __imr void lp_sram_init(void)
 {
 	uint32_t lpsram_ebb_quantity = ace_lpsram_get_bank_count();
 	volatile uint32_t *l2usbpmptr = (volatile uint32_t *)ACE_L2MM->l2usbpmptr;
+	volatile uint8_t *status = (volatile uint8_t *)l2usbpmptr + 4;
```
Comment (kv2019i) [inline soc/xtensa/intel_adsp/ace/sram.c:43]: The 4 smells of a magic number, but we have such magic numbers already in the driver before this PR, so maybe better to change in a follow-up PR.

**Review by kv2019i — DISMISSED** [review]

**Comment by abonislawski** [comment]
Removed z_delay from hpsram init too in optional commit

**Review by kv2019i — DISMISSED** [review]

File: soc/xtensa/intel_adsp/ace/sram.c [hunk 2]
```diff
@@ -28,9 +28,7 @@ __imr void hp_sram_init(uint32_t memory_size)
 		*(l2hsbpmptr + idx * 2) = 0;
 	}
 	for (idx = 0; idx < hpsram_ebb_quantity; ++idx) {
-		while (*(status + idx * 8) != 0) {
-			z_idelay(DELAY_COUNT);
-		}
+		while (*(status + idx * 8) != 0) {}
```
Comment (lgirdwood) [inline soc/xtensa/intel_adsp/ace/sram.c:31]: This can silently spin forever if it fails, can we have a timeout and error code ?

**Review by lgirdwood — COMMENTED** [review]
@abonislawski IIRC, I think all loads/stores are 32bit on Intel arch. If so, this could cause an issue later on, e.g. a CC update.

File: soc/xtensa/intel_adsp/ace/sram.c (hunk 2 above)
Comment (marcinszkudlinski) [reply soc/xtensa/intel_adsp/ace/sram.c:31]: @lgirdwood it is a very basic init and at a very early stage (powering memory banks), can't imagine what can fail here (except extreme problems like HW failure)

**Review by dcpleung — DISMISSED** [review]

**Review by nashif — DISMISSED** [review]

File: soc/xtensa/intel_adsp/ace/sram.c [hunk 3]
```diff
@@ -40,10 +38,15 @@ __imr void lp_sram_init(void)
 {
 	uint32_t lpsram_ebb_quantity = ace_lpsram_get_bank_count();
 	volatile uint32_t *l2usbpmptr = (volatile uint32_t *)ACE_L2MM->l2usbpmptr;
+	volatile uint8_t *status = (volatile uint8_t *)l2usbpmptr + 4;
+	int idx;
```
Comment (ceolin) [inline soc/xtensa/intel_adsp/ace/sram.c:42]: for the sake of correctness this should be `uint32_t` like `lpsram_ebb_quantity`

**Review by dcpleung — APPROVED** [review]

**Comment by abonislawski** [comment]
@nashif checkpatch fixed

**Review by kv2019i — APPROVED** [review]

**Pull request merged**